qiita_ware/ebi.py

```diff
@@ -482,53 +403,81 @@ def generate_experiment_xml(self):
             design = ET.SubElement(experiment, 'DESIGN')
             design_description = ET.SubElement(design,
                                                'DESIGN_DESCRIPTION')
-            design_description.text = escape(clean_whitespace(
-                sample_info['prep']['experiment_design_description']))
+            edd = sample_prep.pop('experiment_design_description')
+            design_description.text = escape(clean_whitespace(edd))
             ET.SubElement(
                 design, 'SAMPLE_DESCRIPTOR', {'refname': sample_alias}
             )
 
-            self._generate_library_descriptor(
-                design, sample_name,
-                sample_info['prep']['library_construction_protocol']
-            )
+            # this is the library contruction section. The only required fields
+            # is library_construction_protocol, the other are optional
+            library_descriptor = ET.SubElement(design, 'LIBRARY_DESCRIPTOR')
+            library_name = ET.SubElement(library_descriptor, 'LIBRARY_NAME')
+            library_name.text = self._get_library_name(sample_name)
+
+            lcp = ET.SubElement(library_descriptor,
+                                "LIBRARY_CONSTRUCTION_PROTOCOL")
+            lcp.text = escape(clean_whitespace(
+                sample_prep.pop('library_construction_protocol')))
+
+            # these are not requiered field but present add them in the right
+            # format
+            if 'library_strategy' in sample_prep:
+                library_strategy = ET.SubElement(library_descriptor,
+                                                 "LIBRARY_STRATEGY")
+                library_strategy.text = sample_prep.pop('library_strategy')
+            if 'library_source' in sample_prep:
+                library_source = ET.SubElement(library_descriptor,
+                                               "LIBRARY_SOURCE")
+                library_source.text = sample_prep.pop('library_source')
+            if 'library_selection' in sample_prep:
+                library_selection = ET.SubElement(library_descriptor,
+                                                  "LIBRARY_SELECTION")
+                library_selection.text = sample_prep.pop('library_selection')
+            if 'library_layout' in sample_prep:
+                library_layout = ET.SubElement(library_descriptor,
+                                               "LIBRARY_LAYOUT")
+                library_layout.text = sample_prep.pop('library_layout')
+                # no idea what SINGLE means
```

Comment:
That's an interesting comment... does anybody know what we have to have single here?